react-state/my-app/src/ToggleButton.tsx

```diff
@@ -0,0 +1,33 @@
+import { useState } from 'react';
+
+type Props = {
+  text: string;
+  color: string;
+  text2: string;
+  color2: string;
+};
+
+export default function ToggleButton({ text, color, text2, color2 }: Props) {
+  let [isClicked, setClicked] = useState(false);
+  console.log('useState:', isClicked);
+
+  function handleClick(): void {
+    console.log('before setter:', isClicked);
+    setClicked(!isClicked);
+    console.log('after setter:', isClicked);
+  }
+
+  if (!isClicked) {
+    return (
+      <button onClick={handleClick} style={{ backgroundColor: color }}>
+        {text}
+      </button>
+    );
+  } else {
+    return (
+      <button onClick={handleClick} style={{ backgroundColor: color2 }}>
+        {text2}
+      </button>
+    );
```

Comment:
It works. I like how clever you were on having two colors and texts. However, I'd like you to look at removing the code duplication before I approve.